lib/util/propTypes.js

```diff
@@ -547,6 +547,35 @@ module.exports = function propTypesInstructions(context, components, utils) {
     if (node.right) return getRightMostTypeName(node.right);
   }
 
+  /**
+   * Returns true if the node is either a interface or type alias declaration
+   * @param {ASTNode} node
+   * @return {boolean}
+   */
+  function filterInterfaceOrTypeAlias(node) {
+    return (
+      astUtil.isTSInterfaceDeclaration(node) || astUtil.isTSTypeAliasDeclaration(node)
+    );
+  }
+
+  /**
+   * Returns true if the interface or type alias declaration node name matches the type-name str
+   * @param {ASTNode} node
+   * @param {string} typeName
+   * @return {boolean}
+   */
+  function filterInterfaceOrAliasByName(node, typeName) {
+    if (node.id && node.id.name) {
+      return node.id.name === typeName;
+    }
+
+    if (node && node.declaration && node.declaration.id && node.declaration.id.name) {
+      return node.declaration.id.name === typeName;
+    }
+
+    return false;
```

Comment:
```suggestion
    return node && (
      (node.id && node.id.name === typeName)
      || (node.declaration && node.declaration.id && node.declaration.id.name === typeName)
    );
```
?

Reply:
sure, but i feel sometimes one liners are less readable.

Reply:
yeah true, in this case it'd be much more readable with optional chaining, but we can't use that yet :-/